front/pages/w/[wId]/builder/data-sources/new.tsx

```diff
@@ -70,6 +72,9 @@ export default function DataSourceNew({
   const [dataSourceDescription, setDataSourceDescription] = useState("");
 
   const formValidation = useCallback(() => {
+    let edited = false;
```

Comment:
NIT: shouldn't `edited` be set to true as long as there was a change on any field, so directly on the onChange methods? I feel it's slightly weird to keep a edited = false if datasource name exists already or starts with manage-.

Reply:
Well you want edited to be false if the value matches the original value, so it's kind of nicer to have that logic centralized?